### PR TITLE
Can we keep the top 3 on the list Knit Crochet [sc-51]

### PR DESCRIPTION
**Pull request by jamesstrate**
- Added helper on model to determine if skill is popular
- Created new partial to render skill assessment form
- populated popular skills first then remaining skills

Screenshot 2023-12-07 at 10 27 05 PM

### CODE REVIEW

File: app/views/finishers/_skills_form.html.haml [hunk 1]
```diff
@@ -7,16 +7,13 @@
...
-                (#{assessment_form.object.skill.description})
-            .col-4.col-md-3.col-lg-2.my-1
-              = assessment_form.hidden_field :skill_id, value: assessment_form.object.skill_id
-              = assessment_form.select :rating, [['', '0'],['Beginner', '1'],['Intermediate', '2'],['Expert', '3']], {}, { class: 'form-select' }
-            .col-4.my-1
-              = assessment_form.text_field :description, class: 'ml-2 form-control', placeholder: "Tell us more..."
+          -if assessment_form.object.skill.is_popular?
+            = render :partial => 'skills_assessment_form', :locals => { skill_assessment_form: assessment_form, featured: true }
+
+        = form.fields_for :assessments, @assessments do |assessment_form|
```
Comment (swards) [inline app/views/finishers/_skills_form.html.haml:13]: We an avoid looping through these again

File: app/models/skill.rb [hunk 2]
```diff
@@ -7,4 +7,8 @@ class Skill < ApplicationRecord
   def self.[](val)
     { 0 => :no_experience, 1 => :beginner, 2 => :intermediate, 3 => :expert }[val]
   end
+
```
[the remark this answers is not in the document]
Comment (jamesstrate) [reply app/models/skill.rb:12]: I like the first option one less step to have to map the name to primary id when adding new. also made it a constant.   Great recommendation!

File: app/controllers/admin/skills_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class Admin::SkillsController < Admin::AdminController
   def index
-    @skills = Skill.all.order(:name)
+    @skills = Skill.all.sorted_by_popularity
```
Comment (swards) [inline app/controllers/admin/skills_controller.rb:3]: Still has the `all`

**Review by swards — APPROVED** [review]

File: app/controllers/admin/skills_controller.rb (hunk 3 above)
Comment (jamesstrate) [reply app/controllers/admin/skills_controller.rb]: oops.  good catch

**Pull request merged**